api/v1alpha1/wildflyserver_types.go

```diff
@@ -67,6 +67,40 @@ type WildFlyServerSpec struct {
 	// ResourcesSpec defines the resources used by the WildFlyServer, ie CPU and memory, use limits and requests.
 	// More info: https://pkg.go.dev/k8s.io/api@v0.18.14/core/v1#ResourceRequirements
 	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
+	// LivenessProbe defines the periodic probe of container liveness. Container will be restarted if the probe fails
+	LivenessProbe *ProbeSpec `json:"livenessProbe,omitempty"`
+	// ReadinessProbe defines the periodic probe of container service readiness. Container will be removed from service endpoints if the probe fails.
+	ReadinessProbe *ProbeSpec `json:"readinessProbe,omitempty"`
+	// StartupProbe indicates that the Pod has successfully initialized. If specified, no other probes are executed until this completes successfully.
+	// If this probe fails, the Pod will be restarted, just as if the livenessProbe failed. This can be used to provide different probe parameters at the beginning of a Pod's lifecycle,
+	// when it might take a long time to load data or warm a cache, than during steady-state operation.
+	StartupProbe *ProbeSpec `json:"startupProbe,omitempty"`
+}
+
+// ProbeSpec Probe describes a health check to be performed against a container to determine whether it is alive or ready to receive traffic.
+// +k8s:openapi-gen=true
+type ProbeSpec struct {
+	// Number of seconds after the container has started before liveness probes are initiated.
+	// Defaults to 60 seconds. Minimum value is 0.
+	// +optional
+	InitialDelaySeconds int32 `json:"initialDelaySeconds,omitempty"`
```

Comment:
it is missing a `kubebuilder` instruction to exert the constraints:

`// +kubebuilder:validation:Minimum=0`

Reply:
I've added the missing `// +kubebuilder:validation:Minimum=0`